coordinator/src/main/scala/filodb.coordinator/QueryActor.scala

```diff
@@ -246,14 +246,13 @@ final class QueryActor(memStore: TimeSeriesStore,
     }
   }
 
-  def checkTimeout(queryContext: QueryContext, replyTo: ActorRef): Boolean = {
-    // timeout can occur here if there is a build up in actor mailbox queue and delayed delivery
-    val queryTimeElapsed = System.currentTimeMillis() - queryContext.submitTime
-    if (queryTimeElapsed >= queryContext.plannerParams.queryTimeoutMillis) {
-      replyTo ! QueryError(queryContext.queryId, QueryStats(),
-        QueryTimeoutException(queryTimeElapsed, this.getClass.getName))
-      false
-    } else true
+  def checkTimeoutBeforeQueryExec(queryContext: QueryContext, replyTo: ActorRef): Boolean = {
+    val ex = queryContext.checkQueryTimeout(this.getClass.getName, false)
+    ex match {
+      case Some(qte) => replyTo ! QueryError(queryContext.queryId, QueryStats(), qte)
+                        false
+      case None =>      true
+    }
   }
```

Comment:
Do we need this extra method? Won't `QueryActor` wrap the timeout exception in a response, anyways?

Reply:
No it doesn't. The call is outside the catch scope.